base/sharedarray.jl

```diff
@@ -249,7 +249,7 @@ function rand!{T}(S::SharedArray{T})
 end
 
 function randn!(S::SharedArray)
-    f = S->map!(x->randn, S.loc_subarr_1d)
+    f = S->map!(x->randn(), S.loc_subarr_1d)
```

Comment:
I think it's because that variable is a `SubArray`, for which `rand!` and `randn!` were not defined in 0.3.
It could be changed now I suppose.